types/coin.go

```diff
@@ -118,16 +118,25 @@ func (coin Coin) AddAmount(amount Int) Coin {
 // Sub subtracts amounts of two coins with same denom. If the coins differ in denom
 // then it panics.
 func (coin Coin) Sub(coinB Coin) Coin {
+	res, err := coin.SafeSub(coinB)
+	if err != nil {
+		panic(err)
+	}
+
+	return res
+}
+
+func (coin Coin) SafeSub(coinB Coin) (Coin, error) {
```

Comment:
One thing I notice is that SafeSub on coins returns `(Coins, bool)`. Any opinions on bool vs error on the second argument?

Also, let's add a godoc

Reply:
As mentioned in issue we thought of returning error instead of panic, so I thought it would be better to return an error directly instead of bool.

Reply:
Error is preferable IMO

Reply:
Here error makes sense